src/ring_theory/subsemiring.lean

```diff
@@ -704,3 +704,51 @@ def sof_left_inverse {g : S → R} {f : R →+* S} (h : function.left_inverse g
   (sof_left_inverse h).symm x = g x := rfl
 
 end ring_equiv
+
+/-! ### Actions by `subsemiring`s
+
+These are just copies of the definitions about `submonoid` starting from `submonoid.mul_action`.
+The only new result is `subsemiring.module`.
+
+A stronger result is provided elsewhere as `algebra.of_subsemiring` which uses the same scalar
+action.
```

Comment:
So this is for `semiring R`, while `algebra.of_subsemiring` is for `comm_semiring R`? If so, mention that here (and in the `ring` file).

Reply:
Yes, but I mean stronger in the sense that the result is an algebra and not just a module; remembering that mathlib requires the base ring of algebras too be commutative

Reply:
I understand, but maybe reformulate as something like 

> `algebra.of_subsemiring` gives a stronger result if `R` is a commutative semiring [...]

Reply:
```suggestion
When `R` is commutative, `algebra.of_subsemiring` provides a stronger result than those found in
this file, which uses the same scalar action.
```